<!DOCTYPE html>
<html lang="en">
<head>
  
  
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    
    <meta name="author" content="Anna Syme">
    <link rel="canonical" href="https://plant-genomics.github.io/training/modules/papers/">
    <link rel="shortcut icon" href="../../img/favicon.ico">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
    <title>Papers - Plant Genomics Training</title>
    <link href="../../css/bootstrap-3.3.7.min.css" rel="stylesheet">
    <link href="../../css/font-awesome-4.7.0.css" rel="stylesheet">
    <link href="../../css/base.css" rel="stylesheet">
    <link rel="stylesheet" href="../../css/highlight.css">
    <link href="../../css/style.css" rel="stylesheet">
    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
        <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
        <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    <script src="../../js/jquery-3.2.1.min.js"></script>
    <script src="../../js/bootstrap-3.3.7.min.js"></script>
    <script src="../../js/highlight.pack.js"></script>
    
    <base target="_top">
    <script>
      var base_url = '../..';
      var is_top_frame = false;
        
        var pageToc = [
          {title: "Papers", url: "#_top", children: [
          ]},
        ];

    </script>
    <script src="../../js/base.js"></script>
    <script>
        (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
        (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
        })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

        ga('create', 'UA-154956469-1', 'https://plant-genomics.github.io/training/');
        ga('send', 'pageview');
    </script> 
</head>

<body>
<script>
if (is_top_frame) { $('body').addClass('wm-top-page'); }
</script>



<div class="container-fluid wm-page-content">
  <a name="_top"></a>
    

    <h1 id="papers">Papers</h1>
<p>Some useful papers about plant bioinformatics.</p>
<!- paperpile library folder ->

<p>Kersey PJ: <strong>Plant genome sequences: past, present, future.</strong> Curr. Opin. Plant Biol. 2019, 48:1–8.</p>
<p>One Thousand Plant Transcriptomes Initiative: <strong>One thousand plant transcriptomes and the phylogenomics of green plants.</strong> Nature 2019, 574:679–685.</p>
<p>Dominguez Del Angel V, Hjerde E, Sterck L, Capella-Gutierrez S, Notredame C, Vinnere Pettersson O, Amselem J, Bouri L, Bocs S, Klopp C, Gibrat J-F, Vlasova A, Leskosek BL, Soler L, Binzer-Panchal M, Lantz H: <strong>Ten steps to get started in Genome Assembly and Annotation.</strong> F1000Res. 2018, 7.</p>
<p>Ghurye J, Pop M: <strong>Modern technologies and algorithms for scaffolding assembled genomes.</strong> PLoS Comput. Biol. 2019, 15:e1006994.</p>
<p>Paajanen P, Kettleborough G, López-Girona E, Giolai M, Heavens D, Baker D, Lister A, Cugliandolo F, Wilde G, Hein I, Macaulay I, Bryan GJ, Clark MD: <strong>A critical comparison of technologies for a plant genome sequencing project.</strong> Gigascience 2019, 8.</p>
<p>Jung H, Winefield C, Bombarely A, Prentis P, Waterhouse P: <strong>Tools and Strategies for Long-Read Sequencing and De Novo Assembly of Plant Genomes.</strong> Trends Plant Sci. 2019, 24:700–724.</p>
<p>Kyriakidou M, Tai HH, Anglin NL, Ellis D, Strömvik MV: <strong>Current Strategies of Polyploid Plant Genome Sequence Assembly.</strong> Front. Plant Sci. 2018, 9:1660.</p>
<p>Ott A, Schnable JC, Yeh C-T, Wu L, Liu C, Hu H-C, Dalgard CL, Sarkar S, Schnable PS: <strong>Linked read technology for assembling large complex and polyploid genomes.</strong> BMC Genomics 2018, 19:651.</p>
<p>Wendel JF, Jackson SA, Meyers BC, Wing RA: <strong>Evolution of plant genome architecture</strong>. Genome Biol. 2016, 17:37.</p>
<p>Chase MW, Christenhusz MJM, Fay MF, Byng JW, Judd WS, Soltis DE, Mabberley DJ, Sennikov AN, Soltis PS, Stevens PF: <strong>An update of the Angiosperm Phylogeny Group classification for the orders and families of flowering plants: APG IV.</strong> Bot. J. Linn. Soc. 2016, 181:1–20.</p>
<p>Sedlazeck FJ, Lee H, Darby CA, Schatz MC: <strong>Piercing the dark matter: bioinformatics of long-range sequencing and mapping.</strong> Nat. Rev. Genet. 2018, 19:329–346.</p>
<p>Brewer GE, Clarkson JJ, Maurin O, Zuntini AR, Barber V, Bellot S, Biggs N, Cowan RS, Davies NMJ, Dodsworth S, Edwards SL, Eiserhardt WL, Epitawalage N, Frisby S, Grall A, Kersey PJ, Pokorny L, Leitch IJ, Forest F, Baker WJ: <strong>Factors Affecting Targeted Sequencing of 353 Nuclear Genes From Herbarium Specimens Spanning the Diversity of Angiosperms.</strong> Front. Plant Sci. 2019, 10:1102.</p>
<p>Comai L, Maheshwari S, Marimuthu MPA: ​ <strong>Plant centromeres​.</strong> Curr. Opin. Plant Biol. 2017, 36:158–167.</p>
<p>Vicient CM, Casacuberta JM: <strong>Impact of transposable elements on polyploid plant genomes.</strong> Ann. Bot. 2017, 120:195–207.</p>
<p>Wang W, Das A, Kainer D, Schalamun M, Morales-Suarez A, Schwessinger B, Lanfear R: <strong>The draft nuclear genome assembly of Eucalyptus pauciflora: new approaches to comparing de novo assemblies.</strong> bioRxiv 2019:678730.</p>

  <br>
</div>

<footer class="col-md-12 wm-page-content">
  <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a> using <a href="https://github.com/gristlabs/mkdocs-windmill">Windmill</a> theme by Grist Labs.</p>
</footer>

</body>
</html>